packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/DatasetController.java

```diff
@@ -671,33 +675,36 @@ public ResponseEntity<JsonNode> getPreview(
 			@PathVariable("id") final UUID id,
 			@RequestParam("filename") final String filename) {
 
-		try {
-			if (filename.endsWith(".nc")) {
-				return climateDataProxy.previewEsgf(id.toString(), null, null, null);
-			} else {
-				final Optional<PresignedURL> url = datasetService.getDownloadUrl(id, filename);
-				// TODO: This attempts to check the file, but fails to open the file, might need
-				// to write a NetcdfFiles Stream reader
-				try (final NetcdfFile ncFile = NetcdfFiles.open(url.get().getUrl())) {
-					final ImmutableList<Attribute> globalAttributes = ncFile.getGlobalAttributes();
-					for (final Attribute attribute : globalAttributes) {
-						final String name = attribute.getName();
-						final Array values = attribute.getValues();
-						// log.info("[{},{}]", name, values);
-					}
-					return climateDataProxy.previewEsgf(id.toString(), null, null, null);
-				} catch (final IOException ioe) {
-					throw new ResponseStatusException(
-							org.springframework.http.HttpStatus.valueOf(415),
-							"Unable to open file");
-				}
-			}
-		} catch (final Exception e) {
-			final String error = "Unable to get download url";
-			log.error(error, e);
-			throw new ResponseStatusException(
-					org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR,
-					error);
-		}
+		// Currently `climate-data` service can only work on NetCDF files it knows about
+		return ResponseEntity.status(HttpStatus.NOT_IMPLEMENTED).build();
+
+//		try {
```

Comment:
do we still need this? or can we delete

Reply:
It is a future maybe - currently the code as written does not work, but maybe it will?